rackcity/views/pdu_views.py

```diff
@@ -51,6 +55,110 @@ def power_status(request, id):
     )
 
 
+"""
+TODO check that power is in opposite state when performing a toggle
+TODO validate if ports exist/are connected
+"""
+@api_view(['POST'])
+@permission_classes([IsAdminUser])
+def power_on(request):
+    """
+    Turn on power to specified port
+    """
+    data = JSONParser().parse(request)
+    failure_message = ""
+    try:
+        asset = Asset.objects.get(id=data['id'])
+    except Asset.DoesNotExist:
+        failure_message = "No model exists with id=" + str(id)
+        return JsonResponse(
+            {"failure_message": failure_message},
+            status=HTTPStatus.BAD_REQUEST,
+        )
+    power_connections = serialize_power_connections(asset)
+    # Check power is off
+    html = requests.get(pdu_url + get_pdu + get_pdu_status_ext(asset, str(power_connections[data['power_port_number']]['left_right'])))
```

Comment:
it might be worth checking that the keys `'id'` and `'power_port_number'` are in data and failing early if not, and maybe also make sure the power_port_number is a valid port on the asset's model. this is mostly helpful for controlling the error messages that get shown to the user so no key errors get tangled up and throw a 500